libraries/abstractions/common_io/test/test_iot_spi.c

```diff
@@ -1119,10 +1130,6 @@ TEST( TEST_IOT_SPI, AFQP_IotSPI_CancelSuccess )
             TEST_ASSERT_EQUAL( IOT_SPI_SUCCESS, lRetVal );
         }
 
-        /* Wait to make sure operation was really canceled. */
-        xCallbackReturn = xSemaphoreTake( xtestIotSPISemaphore, testIotSPI_DEFAULT_SEMAPHORE_DELAY );
```

Comment:
Thanks for catching this bug. But should we fix it by setting the callback instead?

```
iot_spi_set_callback( xSPIHandle, prvSpiAsyncCallback, NULL );
```

Reply:
I did consider that, but removed it because there is a race condition. If the async spi transaction finishes before the [subsequent call to `iot_spi_cancel`](https://github.com/aws/amazon-freertos/blob/master/libraries/abstractions/common_io/test/test_iot_spi.c#L1115), then the installed callback would have given the semaphore and [the following assertion](https://github.com/aws/amazon-freertos/blob/master/libraries/abstractions/common_io/test/test_iot_spi.c#L1124) would be violated since taking the given sem would be `pdPASS`.

Concluded it'd be better to remove it for now, until more determinism can be applied to avoid the race condition. 
For example, if there was a portable CommonIO way to stop the SPI driver source clock, doing so could likely halt the pending transmission, at least for nrf52.

Reply:
Yeah, the test for cancel is not easy to be deterministic. I am OK with removing it for now.